tests/core/usage_statistics/test_usage_statistics_data_context.py

```diff
@@ -0,0 +1,98 @@
+                ENABLED_METHODS,
+            ),
+        ),
+    )
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@pytest.mark.parametrize(
+    "data_context_fixture_name",
+    [
+        # In order to leverage existing fixtures in parametrization, we provide
+        # their string names and dynamically retrieve them using pytest's built-in
+        # `request` fixture.
+        # Source: https://stackoverflow.com/a/64348247
+        pytest.param("in_memory_runtime_context", id="EphemeralDataContext"),
+        pytest.param("empty_data_context", id="FileDataContext"),
+        pytest.param(
+            "empty_data_context_in_cloud_mode",
+            id="CloudDataContext",
+        ),
+    ],
+)
+@pytest.mark.integration
+def test_all_relevant_context_methods_emit_usage_stats(
+    mock_emit: mock.MagicMock,
+    usage_stats_decorated_methods_on_abstract_data_context: list[str],
+    enable_usage_stats,  # Needs to be before context fixtures to ensure usage stats handlers are attached
+    data_context_fixture_name: str,
+    request,
+):
+    """
+    What does this test and why?
+
+    Ensures that all primary context types (EphemeralDataContext, FileDataContext, and CloudDataContext)
+    emit the same usage stats events.
+
+    This guards against the case where a child class overrides a method defined by AbstractDataContext
+    but forgets to add the decorator, resulting in us losing event data.
+    """
+    context = request.getfixturevalue(data_context_fixture_name)
+
+    relevant_methods = usage_stats_decorated_methods_on_abstract_data_context
+    expected_events = (
+        UsageStatsEvents.DATA_CONTEXT_ADD_DATASOURCE,
+        UsageStatsEvents.DATA_CONTEXT_BUILD_DATA_DOCS,
+        UsageStatsEvents.DATA_CONTEXT_GET_BATCH_LIST,
+        UsageStatsEvents.DATA_CONTEXT_OPEN_DATA_DOCS,
+        UsageStatsEvents.DATA_CONTEXT_RUN_CHECKPOINT,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_ON_DATA,
+        UsageStatsEvents.DATA_CONTEXT_RUN_RULE_BASED_PROFILER_WITH_DYNAMIC_ARGUMENTS,
+        UsageStatsEvents.DATA_CONTEXT_RUN_VALIDATION_OPERATOR,
+        UsageStatsEvents.DATA_CONTEXT_SAVE_EXPECTATION_SUITE,
+    )
+    assert len(relevant_methods) == len(
+        expected_events
+    ), "Please update the `expected_events` list to account for all usage stats decorated methods in AbstractDataContext"
+
+    for method_name, expected_event in zip(relevant_methods, expected_events):
+        logger.info(f"Testing {context.__class__}.{method_name}")
+
+        # As we only care about the decorator and not the underlying method being decorated,
+        # we use the following try/except pattern. All method calls will generally fail due
+        # to having no input args but we still manage to trigger our target decorator.
+        try:
+            method = getattr(context, method_name)
+            method()
+        except Exception:
+            pass
```

Comment:
This pattern unfortunately results in a side effect - `build_data_docs` runs through and actually opens a site in the browser. We need to determine a better way to do this.

Perhaps I could mock every function except the ones decorated above?

Reply:
Can you save the event with the method name in `ENABLED_METHODS`. Then you can check directly that the method is associated with the correct event without calling the method.

You could then have a separate test for the decorator where you wrap a no-op function with the decorator configured however, and test you see the correct event emitted. That is you have a separate test that the decorator works and this test ensures the right event is associated with the right method.